src/azul/indexer/__init__.py

```diff
@@ -112,34 +108,27 @@ def parse(cls, prefix: str) -> 'Prefix':
 
         >>> list(Prefix.parse('8f538f53/0').partition_prefixes())
         ['']
+
+        >>> Prefix.parse('')
+        Traceback (most recent call last):
+        ...
+        azul.RequirementError: Cannot parse an empty prefix source
         """
         source_delimiter = '/'
+        reject(prefix == '', 'Cannot parse an empty prefix source')
         reject(prefix.endswith(source_delimiter),
                'Prefix source cannot end in a delimiter.', prefix, source_delimiter)
-        if prefix == '':
-            entry = ''
-            partition = None
-        else:
-            try:
-                entry, partition = prefix.split(source_delimiter)
-            except ValueError:
-                entry = prefix
-                partition = None
-            if partition:
-                try:
-                    partition = int(partition)
-                except ValueError:
-                    raise ValueError('Partition prefix length must be an integer.', partition)
+        try:
+            entry, partition = prefix.split(source_delimiter)
+        except ValueError:
+            raise ValueError('Missing partition prefix length.', prefix)
```

Comment:
The other error messages in this class do not end with a period:
```suggestion
            raise ValueError('Missing partition prefix length', prefix)
```